### PR TITLE
Feature/new booking module

### PR DESCRIPTION
**Pull request by MoShrank**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   "dependencies": {
     "@apollo/client": "^3.0.0-rc.5",
     "@sentry/browser": "^5.15.5",
+    "@types/resize-observer-browser": "^0.1.4",
```
Comment (MathisBullinger) [inline package.json]: Please move to `devDependencies`

File: src/components/BookingDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+export default function Dropdown({ values, onSelect, selectedValue }: Props) {
+  const [valuesVisible, setValuesVisible] = useState(false)
+  const valuesRef = useRef(null)
+  useDetectClickOutside(valuesRef, () => setValuesVisible(false))
+
+  return (
+    <Style.DropDownContainer ref={valuesRef}>
+      <Style.DropDownHeader onClick={() => setValuesVisible(!valuesVisible)}>
+        {selectedValue}
+        <Icon color={'rgba(0, 0, 0, 0.46)'} icon="drop_down_arrow" />
```
Comment (MathisBullinger) [inline src/components/BookingDropdown.tsx]: `color={'…'}` should be rewritten as `color="…"` in JSX, but why is color applied as an inline style? This should be included in the regular CSS.

File: src/screens/Mentor/Request.jsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect } from 'react'
+import { mutations, useMutation } from 'gql'
+import { Title, Textbox, Button, Icon } from '../../components/'
+import { notify } from 'notification'
+import { useMe } from 'utils/hooks'
+import { ordNum } from '../../utils/date'
+import styled from 'styled-components'
+
+const WEEK_DAYS = [
```
Comment (MathisBullinger) [inline src/screens/Mentor/Request.jsx]: That does not look like the days of the week to me…

File: src/screens/Mentor/Request.jsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+]
+
+const formatAMPM = date => {
```
Comment (MathisBullinger) [inline src/screens/Mentor/Request.jsx]: The browser can do that for you, no need to format the time manually into AM/PM

```js
date.toLocaleTimeString('en-US', { hour: 'numeric', minute: '2-digit' })
```

File: src/screens/Mentor/Request.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    <Styles.SchedulingContainer>
+      <Styles.MessageBoxContainer>
+        <Title size={4}>Message</Title>
+        <Textbox
+          placeholder="I have challenge x and was hoping you could help me with y."
+          values={msg}
+          onChange={setMsg}
+        />
+        <Styles.Hint>
+          We’ll send an invittion to Tocha. You’ll receive a confirmation for
```
Comment (MathisBullinger) [inline src/screens/Mentor/Request.jsx]: `invittion` typo and not every mentor's name is Tocha ;)

**Review by MathisBullinger — CHANGES_REQUESTED** [review]
I merged the `staging` branch into the `feature/new-booking-module` branch. There were some conflicts that I had to settle. 
We both added a `Dropdown` component. For now, I just moved your Dropdown component to `components/BookingDropdown.tsx`. But at some point we should combine all those components (there now are at least 5 different dropdown components that I now of of the top of my head).
We also both added a `calendar` and `clock` icon, so I had to pick one of each, please see if they work for you.

Besides the comments that I made in the code a few general things:

- please run the `npm run lint` script—*I updated it to include all the `.ts` & `.tsx` files*—and fix all the issues that it brings up. I.e. there shouldn't be any unused imports, logging out random variables, or missing dependencies from `useEffect` hook dependency arrays. If an eslint rule should not apply somewhere, you can disable it with `// eslint-disable-next-line <name-of-rule>`, but please only do so if there is a good reason for the exception. I disabled the stylelint `value-keyword-case` rule, it caused the `something dummyValue` errors in the styled-components section, and disabling it isn't too big of a loss imo anyway
- if there is a missing type annotation every once in a while, it's no big deal. But please don't leave them out categorically. Without them, typescript isn't half as useful. Also, the javascript files that still exist in the frontend are only there because they haven't been worked on since we started using typescript. So please convert the files in the `screens/Mentor/` directory to typescript files (using `git mv`, so VCS doesn't get upset)
- you can use import paths relative to the `src` dir. I.e. `../../../utils/something` can be imported as `utils/something`
- we have the convention of using `rem` units—or `em`s, where appropriate—instead of `px`. So if we decide to change the root font size at some point, it cascades to all UI elements. You still can use `px` to specify borders and other properties that should only be a line without any area. If you don't want to, you don't have to change all the units for the booking module, but please keep it in mind for future code
- JSX attributes like `foo={'some_constant_string'}` can—and should—be written as `foo="some_constant_string"`

File: src/components/BookingDropdown.tsx (hunk 2 above)
Comment (MoShrank) [reply src/components/BookingDropdown.tsx]: It's not an inline style but a property to the Icon component

Comment (MathisBullinger) [reply src/components/BookingDropdown.tsx]: …which then applies it as an inline style

**Review by MathisBullinger — APPROVED** [review]

**Pull request merged**